src/main/java/dev/jbang/ExtendedRunUnit.java

```diff
@@ -0,0 +1,269 @@
+package dev.jbang;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class wraps a reference to an runnable/executable resource in the form
+ * of a RunUnit. It also holds all additional information necessary to be able
+ * to actually run/execute that resource. This is information that can not be
+ * induced or extracted from the resource itself but is information that is
+ * provided by the user or by the environment at runtime.
+ *
+ * This class also implements RunUnit, passing all calls directly to the wrapped
```

Comment:
I kinda feel that these are different things and should use delegation rather than inheritance ...

Reply:
Well it uses delegation. That's why RunUnit is an interface. So the ExtendedRunUnit delegates to the wrapped RunUnit (Being Script or Jar)